### ISSUE TITLE
Device randomly freezes

### ISSUE THREAD

**Issue by Huahddd** [koreader/koreader#12818]
* KOReader version: 2024.11
* Device: kobo forma

#### Issue
When using Koreader, the device often freezes. I asked about this issue and someone told me that it was a defect of the Kobo Forma device. But later I found that when using Kobo's original system, the device rarely freezes, and the original system is more stable than Koreader. So does Koreader have some kind of defect?

**Comment by pazos** [comment]
Nope. Your device is dying. Probably storage related. Use nickel: it is simpler, has less features. You can't avoid issues but at least you can reduce their ocurrence (as stated in your ` I found that when using Kobo's original system, the device rarely freezes`).

If the device is under warranty return it.

**Comment by Huahddd** [comment]
> An issue which still features [a patch](https://github.com/koreader/koreader/issues/11511#issuecomment-1979345163) nobody ever got back to me about (hint, hint).一个问题仍然具有一个[补丁](https://github.com/koreader/koreader/issues/11511#issuecomment-1979345163)，没有人回复我（提示，提示）。

I saw that patch, but I don't know how to use it. Can you tell me how to use that patch? I'll try it and let you know the result

**Comment by Frenzie** [comment]
It shows which lines to change in which files. + is added, - is removed.

**Comment by Huahddd** [comment]
> It shows which lines to change in which files. + is added, - is removed.它显示哪些文件中要更改的行。+ 被添加，- 被删除。

I don't know anything about programming, and I don't know how to use that patch

**Comment by Frenzie** [comment]
Try these files (in `/mnt/onboard/.adds/koreader/ffi`)

[framebuffer_.zip](https://github.com/user-attachments/files/17966469/framebuffer_.zip)

**Comment by Huahddd** [comment]
> Try these files (in `/mnt/onboard/.adds/koreader/ffi`)
> 
> [framebuffer_.zip](https://github.com/user-attachments/files/17966469/framebuffer_.zip)

I have used this file, I will try and report back to you.
Also, if I need to replace the file again when updating koreader

**Comment by NiLuJe** [comment]
Yes, updating will overwrite these changes.

**Comment by Huahddd** [comment]
> Yes, updating will overwrite these changes.是的，更新将覆盖这些更改。

You are awesome, this patch solved my problem perfectly. I read for about 2 hours a day and after using this patch, I haven't had a single freeze.

**Comment by NiLuJe** [comment]
Welp, guess I'll add that one to the "winter break" pile ;).

Do get back to us in a few weeks, in case that last week just happened to be a happy coincidence ;).
